shared/src/commonMain/kotlin/org/hyperskill/app/step_quiz/presentation/StepQuizActionDispatcher.kt

```diff
@@ -53,10 +56,12 @@ class StepQuizActionDispatcher(
                             val message = getSubmissionState(attempt.id, action.step.id, currentProfile.id).fold(
                                 onSuccess = {
                                     Message.FetchAttemptSuccess(
-                                        action.step,
-                                        attempt,
-                                        it,
-                                        isProblemsLimitReached
+                                        step = action.step,
+                                        attempt = attempt,
+                                        submissionState = it,
+                                        isProblemsLimitReached = isProblemsLimitReached,
+                                        shouldShowParsonsModal = action.step.block.name == BlockName.PARSONS &&
+                                            !onboardingInteractor.isParsonsOnboardingShown()
```

Comment:
Let's pass just a isParsonsOnboardingShown flag.
In my opinion, it's better to move step-type detection to reducer.